### PR TITLE
ValuesJsonConverter.cs refactoring

### PR DESCRIPTION
**Pull request by kirk-marple**
@RehanSaeed I've refactored ValuesJsonConverter.cs a good bit.

This passes all unit tests, and even better, doesn't hit your exception 'catch' statements for any of the unit tests. Note some of the // REVIEW comments.

I may have taken liberties with reusing common refactored methods between the single type, and multiple type arg, use cases, but it seems to work.

NOTE: I've kept same semantics where it picks the *last* matching type in the list, by going right-to-left and selecting first one that matches.  This should be a perf benefit, since it was doing wasted work before.

### CODE REVIEW

File: Source/Schema.NET/ValuesJsonConverter.cs [hunk 1]
```diff
@@ -212,6 +161,216 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
         public virtual void WriteObject(JsonWriter writer, object value, JsonSerializer serializer) =>
             serializer.Serialize(writer, value);
 
+        private static object ParseTokenArguments(JToken token, JsonToken tokenType, Type type, object value)
+        {
+            const int SCHEMA_ORG_LENGTH = 18;
```
Comment (RehanSaeed) [inline Source/Schema.NET/ValuesJsonConverter.cs]: Can we please add a comment about what this length is?

Comment (kirk-marple) [reply Source/Schema.NET/ValuesJsonConverter.cs]: Sure, I just flattened that "inline string".Length into a constant.

**Comment by kirk-marple** [comment]
> This looks really promising. Thank you for your contribution. Are there any further tests we could add to make sure that this is working as expected?

I'll put some thought into this.  Have to compare all possible combinations of values/types and make sure we have tests to cover them.

I'm using MusicRecording, MusicGroup, MusicEvent, VideoObject, MusicVenue (and just Thing) in my own app, and not seeing it hit any exceptions with my use cases either.

I think the big change was ignoring string to interface/class casting; that was causing a lot of the exceptions.

**Comment by kirk-marple** [comment]
@RehanSaeed Just added tests for MusicRecording, MusicGroup, MusicEvent, MusicVenue, MusicAlbum.  All passed w/o any further changes.  And added the comment you suggested.  I'd say this is good to go.

**Comment by RehanSaeed** [comment]
Thank you for your contribution. Added you to the front page.

**Comment by RehanSaeed** [comment]
Also thanks for the extra tests, they are beautiful.

**Pull request merged**